### ISSUE TITLE
get_device_setting may return None

### ISSUE THREAD

**Issue by benjaoming** [learningequality/kolibri#6483]
### Observed behavior

We received a traceback in a log file (see below).

@rtibbles notes:

> for the latter file that was posted looks like my refactor of the device settings code has introduced a regression - it doesn't seem to be returning the default when the settings are `None` inside `get_device_setting`

### Expected behavior

API works

### User-facing consequences

Not sure

### Errors and logs

```
ERROR 2020-01-20 09:14:54,827 exception Internal Server Error: /api/
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "./kolibri/core/device/middleware.py", line 60, in __call__
    path_valid = is_valid_path(language_path, urlconf)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/base.py", line 158, in is_valid_path
    resolve(path, urlconf)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/base.py", line 27, in resolve
    return get_resolver(urlconf).resolve(path)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/resolvers.py", line 366, in resolve
    sub_match = pattern.resolve(new_path)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/resolvers.py", line 366, in resolve
    sub_match = pattern.resolve(new_path)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/resolvers.py", line 366, in resolve
    sub_match = pattern.resolve(new_path)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/resolvers.py", line 366, in resolve
    sub_match = pattern.resolve(new_path)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/resolvers.py", line 366, in resolve
    sub_match = pattern.resolve(new_path)
  File "/usr/lib/python3/dist-packages/kolibri/dist/django/urls/resolvers.py", line 361, in resolve
    match = self.regex.search(path)
  File "./kolibri/core/device/translation.py", line 154, in regex
    device_language = get_device_language() or get_settings_language()
  File "./kolibri/core/device/translation.py", line 25, in get_device_language
    language_id = get_device_setting("language_id", None)
  File "./kolibri/core/device/utils.py", line 25, in get_device_setting
    return getattr(device_settings, setting)
AttributeError: 'NoneType' object has no attribute 'language_id'
```

### Steps to reproduce

n/a

### Context

A complicated upgrade with migration issues affecting the `device_devicesettings` table

**Comment by ralphiee22** [comment]
After looking at the code, I am wondering how did the cached value get set to None, as it appears there are safeguards https://github.com/learningequality/kolibri/blob/release-v0.13.x/kolibri/core/device/models.py#L36.
Other than that, `getattr` should have not been called with a `None` object.

**Comment by benjaoming** [comment]
»If something can be assigned None, it will at some point be assigned None«
- benjaoming, January 22 2020 :)

But yeah, that's also a good question. I think it's okay to just say "it's complicated" since it followed from a series of broken upgrade attempts.. processes killed, migrations half run etc.

**Comment by indirectlylit** [comment]
There are others who would argue that if a bug hasn't been correctly explained, it hasn't actually been addressed.

The real truth is somewhere in the middle and usually has to do with deadlines and priorities.

**Comment by rtibbles** [comment]
Fixed in #6517
